examples/huggingface/pytorch/text-generation/quantization/run_generation.py

```diff
@@ -58,198 +55,93 @@
         }
     else:
         op_type_dict = {}
     excluded_precisions = [] if args.int8_bf16_mixed else ["bf16"]
-    if args.sq:
-        args.alpha = args.alpha if args.alpha == "auto" else float(args.alpha)
-        recipes = {"smooth_quant": True, "smooth_quant_args": {"alpha": args.alpha}}
-        conf = PostTrainingQuantConfig(
-            backend="ipex" if args.ipex else "default",
-            excluded_precisions=excluded_precisions,
-            op_type_dict=op_type_dict,
-            recipes=recipes,
-            example_inputs=example_inputs,
-        )
-    else:
-        conf = PostTrainingQuantConfig(
-            backend="ipex" if args.ipex else "default",
-            excluded_precisions=excluded_precisions,
-            op_type_dict=op_type_dict,
-            example_inputs=example_inputs,
-        )
-    # save config
-    user_model.config.save_pretrained(args.output_dir)
-    q_model = quantization.fit(
-        user_model,
-        conf,
-        calib_func=calib_func,
-    )
-    q_model.save(args.output_dir)
+    sq_config = SmoothQuantConfig(
+                                tokenizer=tokenizer,  # either two of one, tokenizer or calib_func
+                                alpha=float(args.alpha),    # default is 0.5
+                                op_type_dict=op_type_dict,  # default is {}
+                                excluded_precisions=excluded_precisions,  # default is []
+                               )
+    # smooth-quant
+    q_model = AutoModelForCausalLM.from_pretrained(args.model,
+                                                   quantization_config=sq_config
+                                               )
+    print("sq done.")
+    # weight-only
+    woq_config = WeightOnlyQuantizationConfig(algorithm="RTN", # default is "RTN"
+                                              bits=8, # default is 8
+                                              group_size=-1, # default is -1
+                                              scheme="sym", # default is sym
+                                              enable_full_range=True # default is True
+                                              ) 
+    woq_model = AutoModelForCausalLM.from_pretrained(args.model,
+                                                quantization_config=woq_config
+                                            )
+    print("woq done.")
+    # amp
+    amp_config = AMPConfig(dtype="bfloat16") # default is bfloat16
+    amp_model = AutoModelForCausalLM.from_pretrained(args.model,
+                                                quantization_config=amp_config
+                                            )
+    print("amp done.")
+    # bitsandbytes
+    bab_config = BitsAndBytesConfig()
+    bab_model = AutoModelForCausalLM.from_pretrained(args.model,
+                                                quantization_config=bab_config
+                                            )
+    print("bitsandbytes done.")

```

Comment:
Why we need to use these configs/generate these models one by one? they should be enabled based on parameter?

Reply:
it seems not only have generate utility but also have dataloader and distribute utility. @hshen14 @PenghuiCheng